external_agent/examples/agent_builder/utils.py

```diff
@@ -99,27 +105,53 @@ async def get_llm_stream(messages: List[Message], thread_id: str):
             WATSONX_DEPLOYMENT_ID, payload
         ):
             logger.info(f"Received chunk from AI service: {chunk}")
-            result = json.loads(chunk)["choices"][0]["message"]
-
-            if result["role"] != "assistant" or "delta" not in result:
-                continue
+            try:
+                delta = json.loads(chunk)["choices"][0]["delta"]
+            except:
+                warning_msg = "It's likely that you are using an old AI service built with a wrong/obsolete response schema, please deploy a new AI service."
+                logger.warning(warning_msg)
+                raise RuntimeError(warning_msg)
 
             current_timestamp = int(time.time())
+            delta_id = str(uuid.uuid4())
             struct = {
-                "id": str(uuid.uuid4()),
-                "object": "thread.message.delta",
+                "id": delta_id,
+                "object": "thread.run.step.delta",
                 "created": current_timestamp,
                 "thread_id": thread_id,
-                "model": "wx.ai AI service",
-                "choices": [
-                    {
+                "model": "wx.ai AI service"
+            }
+            if delta["role"] == "assistant" and "tool_calls" in delta:
+                struct["choices"] = [{
                         "delta": {
-                            "content": result["delta"],
                             "role": "assistant",
+                            "step_details": {
+                                "type": "tool_calls",
+                                "tool_calls": [
+                                    {
```

Comment:
do you have a matching `id` in the json to match the `tool_call_id` in tool_response?

Reply:
thanks for the review Haode! Good catch! let me add that

Reply:
tested new commit again in staging
<img width="797" alt="image" src="https://github.com/user-attachments/assets/17a64d0c-79fd-483d-a02d-f4d9d1ab8981" />